src/v2/Apps/Auction/Routes/Register/index.tsx

```diff
@@ -97,33 +108,35 @@ export const RegisterRoute: React.FC<RegisterProps> = props => {
     }
 
     trackRegistrationFailed(errorMessages)
-
-    actions.setSubmitting(false)
     actions.setStatus("submissionFailed")
   }
 
-  function handleSubmit(actions: FormikActions<object>, result: FormResult) {
-    createCreditCardAndUpdatePhone(
-      relay.environment,
-      result.phoneNumber,
-      result.token.id
-    )
-      .then(() => {
-        createBidder()
-          .then((data: RegisterCreateBidderMutationResponse) => {
-            trackRegistrationSuccess(data.createBidder.bidder.internalID)
-
-            window.location.assign(
-              `${sd.APP_URL}/auction/${sale.slug}/confirm-registration`
-            )
-          })
-          .catch(error => {
-            handleMutationError(actions, error)
-          })
-      })
-      .catch(error => {
-        handleMutationError(actions, error)
-      })
+  async function createTokenAndSubmit(
+    values: FormValuesForRegistration,
+    actions: FormikActions<FormValuesForRegistration>
+  ) {
+    const address = toStripeAddress(values.address)
+    const { phoneNumber } = values.address
+    const { setFieldError, setSubmitting } = actions
+
+    try {
```

Comment:
Is it correct to say that this block is the crux of the behavior change introduced by this PR? By only catching errors in one place, we ensure we're handling them a consistent way?

Reply:
Correct. We could make the behaviour consistent without consolidating these two functions, but I think it's simpler when we do both in one function.